File: A035513/A035513.py
# Wythoff array read by antidiagonals.
# https://oeis.org/A035513

import math

# Applying the fibonnaci rule for making rows.
def fib_rule(num1, num2, n):
    mem = [None]*(n+2)
    mem[0] = num1
    mem[1] = num2
    for i in range(2, n + 2):
        mem[i] = mem[i-1] + mem[i-2]
    return mem[2:]

# Makes the initial two columns necessary to build the sequence.
def two_columns(n):
    first_column = [i for i in range(n)]
    second_column = [math.floor(i * ((1 + math.sqrt(5))/2)) for i in range(1, n + 1)]
    return [first_column, second_column]

# Makes a matrix out of the combination of the two columns and the fibonacci rule.
def make_warray(n):
    columns_array = two_columns(n)
    mtx = []
    for i in range(n):
        row = fib_rule(columns_array[0][i], columns_array[1][i], n - i)
        mtx.append(row)
    return mtx

# Reads the matrix anti-diagonally.
def anti_d_parser(n):
    mtx = make_warray(n)
    sequence = []
    for k in range(0, n):
        j = k
        i = 0
        while (j >= 0):
            sequence.append(mtx[i][j])
            j -= 1
            i += 1
    return sequence

# Prints as an array, the number 8 refers to the elements of the first row.
print(anti_d_parser(8))

File: A035513/A035513.cpp
// Wythoff array read by antidiagonals.
// https://oeis.org/A035513

# include <iostream>
# include <cmath>
# include <vector>
using namespace std;

// Applying the fibonacci rule in making rows.
vector<int> fib_rule(int num1, int num2, int n)
{
   vector<int> mem;
   mem.push_back(num1 + num2);
   mem.push_back(mem[0] + num2);
   for (int i = 2; i < n; i++)
   {
      mem.push_back(mem[i-1] + mem[i-2]);
   }
   return mem;
}

// Makes the initial two columns necessary to build the sequence.
vector< vector<int> > two_columns(int n)
{
   vector< vector<int> > otp;
   vector<int> first_column;
   for (int i = 0; i < n; i++)
   {
      first_column.push_back(i);
   }
   vector<int> second_column;
   for (int i = 1; i <= n; i++)
   {
      second_column.push_back(floor(i * ((1 + sqrt(5))/2)));
   }
   otp.push_back(first_column);
   otp.push_back(second_column);
   return otp;
}

// Makes a matrix out of the combination of the two columns and the fibonacci rule.
vector< vector<int> > make_warray(int n)
{
   vector< vector<int> > columns_array = two_columns(n);
   vector< vector<int> > mtx;
   for (int i = 0; i < n; i++)
   {
      vector<int> row = fib_rule(columns_array[0][i], columns_array[1][i], n - i);
      mtx.push_back(row);
   }
   return mtx;
}

// Reads the matrix anti-diagonally.
vector<int> anti_d_parser(int n)
{
   vector< vector<int> > mtx = make_warray(n); 
   vector<int> sequence;
   for (int k = 0; k < n; k++)
   {
      int j = k;
      int i = 0;
      while (j >= 0)
      {
         sequence.push_back(mtx[i][j]);
         j -= 1;
         i += 1;
      }
   }
   return sequence;
}

int main()
{
   // Prints the matrix as an array, the number 8 refers to the number of elements of the first row.
   vector<int> otp = anti_d_parser(8);

   // Prints the seqeunce, notice that the number of elements in the array is given by the formula: N + (N - 1) + (N - 2) + ... + 1, where N is the number of elements in the first row.
   for (int i = 0; i < 36; i++)
   {
      cout << otp[i] << ' ';
   }
   cout << '\n';
}
